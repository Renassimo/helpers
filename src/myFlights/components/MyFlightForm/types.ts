import { MyFlightAttributes } from '@/myFlights/types';
import { Matcher } from '@/common/types/matchers';

export default interface MyFlightFormFieldProps {
  name: keyof MyFlightAttributes;
  label: string;
  options?: string[];
  matchers?: Matcher;
  type?: string;
  disabled?: boolean;
  isDate?: boolean;
}
