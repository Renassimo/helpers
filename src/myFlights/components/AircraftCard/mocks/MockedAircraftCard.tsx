const MockedAircraftCard = jest.fn(({ chosen }) => (
  <div>MockedAircraftCard - {chosen && 'chosen'}</div>
));

export default MockedAircraftCard;
