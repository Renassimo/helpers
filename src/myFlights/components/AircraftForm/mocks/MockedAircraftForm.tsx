const MockedAircraftForm = jest.fn(({ registration }) => (
  <div>MockedAircraftForm - {registration}</div>
));

export default MockedAircraftForm;
