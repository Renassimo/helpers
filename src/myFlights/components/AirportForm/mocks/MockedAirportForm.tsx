const MockedAirportForm = jest.fn(({ title }) => (
  <div>MockedAirportForm - {title}</div>
));

export default MockedAirportForm;
