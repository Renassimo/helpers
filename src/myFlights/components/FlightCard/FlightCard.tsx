import { Avia } from '@/avia/types/avia';

import AviaCard from '@/avia/components/AviaCard';

const FlightCard = ({
  data,
  onClick,
  chosen = false,
}: {
  data: Avia.FlightData;
  onClick: () => void;
  chosen?: boolean;
}) => {
  const { attributes } = data;
  const {
    registration,
    distance,
    airline,
    aircraft,
    origin,
    destination,
    originName,
    destinationName,
    photoUrl,
    flightNumber,
  } = attributes;

  const withRoute = origin || destination;
  const withNamedRoute = originName || destinationName;

  const additionalContent = [
    withNamedRoute && `${originName} - ${destinationName}`,
    withRoute && `${origin} - ${destination}`,
    airline,
    aircraft,
    registration,
    distance ? `${distance} km` : null,
  ];

  const link =
    flightNumber && `https://www.google.com/search?q=${flightNumber}`;

  return (
    <AviaCard
      onClick={onClick}
      chosen={chosen}
      imageUrl={photoUrl ?? '/images/plane.png'}
      imageAlt={flightNumber}
      link={link}
      title={flightNumber}
      additionalContent={additionalContent}
    />
  );
};

export default FlightCard;
