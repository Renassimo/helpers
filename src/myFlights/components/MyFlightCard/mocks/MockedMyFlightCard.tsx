const MockedMyFlightCard = jest.fn(({ data }) => (
  <div>MockedMyFlightCard - {data?.id}</div>
));

export default MockedMyFlightCard;
