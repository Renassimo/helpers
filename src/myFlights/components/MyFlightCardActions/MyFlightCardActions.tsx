import Grid from '@mui/material/Grid';
import IconButton from '@mui/material/IconButton';

import SearchIcon from '@mui/icons-material/Search';
import OpenInNewIcon from '@mui/icons-material/OpenInNew';
import EditInNewIcon from '@mui/icons-material/Edit';

import useThemeBreakpoints from '@/common/hooks/useThemeBreakpoints';

import { MyFlightData } from '@/myFlights/types';

const MyFlightCardActions = ({ data }: { data: MyFlightData }) => {
  const {
    down: { md: densed },
  } = useThemeBreakpoints();

  const { attributes } = data;
  const { planespottersUrl, registration, url } = attributes;

  const searchLink =
    planespottersUrl ??
    (registration &&
      `https://www.planespotters.net/search?q=${registration}`) ??
    '';

  return (
    <Grid container direction={densed ? 'row' : 'column'}>
      <Grid item>
        <IconButton aria-label="Edit" onClick={() => console.log('Edit')}>
          {<EditInNewIcon />}
        </IconButton>
      </Grid>
      <Grid item>
        <IconButton
          aria-label="Open in Notion"
          LinkComponent={'a'}
          href={url}
          target="_blank"
        >
          {<OpenInNewIcon />}
        </IconButton>
      </Grid>
      {searchLink && (
        <Grid item>
          <IconButton
            aria-label="Search in Planespotters"
            LinkComponent={'a'}
            href={searchLink}
            target="_blank"
          >
            {<SearchIcon />}
          </IconButton>
        </Grid>
      )}
    </Grid>
  );
};

export default MyFlightCardActions;
