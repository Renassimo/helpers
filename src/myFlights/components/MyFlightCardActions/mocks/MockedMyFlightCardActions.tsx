const MockedMyFlightCardActions = jest.fn(({ data }) => (
  <div>MockedMyFlightCardActions - {data?.id}</div>
));

export default MockedMyFlightCardActions;
