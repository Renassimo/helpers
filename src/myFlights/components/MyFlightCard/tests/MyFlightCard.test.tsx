import renderWithTheme from '@/common/tests/helpers/renderWithTheme';

import useThemeBreakpoints from '@/common/hooks/useThemeBreakpoints';
import { showWhen } from '@/common/utils/dayjs';

import MyFlightCardActions from '@/myFlights/components/MyFlightCardActions';

import MockedMyFlightCardActions from '@/myFlights/components/MyFlightCardActions/mocks';
import { mockedMyFlight, mockedMyFlight3 } from '@/myFlights/types/mocks';

import FlightCard from '../MyFlightCard';

jest.mock('@/common/hooks/useThemeBreakpoints');
jest.mock('@/common/utils/dayjs');
jest.mock('@/myFlights/components/MyFlightCardActions');

describe('MyFlightCard', () => {
  const mockedBreakpoints = {
    down: { sm: false },
  };
  const mockedUseThemeBreakpoints = jest.fn(() => mockedBreakpoints);

  const mockedShowWhen = jest.fn((arg) => arg);

  beforeEach(() => {
    (useThemeBreakpoints as unknown as jest.Mock).mockImplementation(
      mockedUseThemeBreakpoints
    );
    (showWhen as unknown as jest.Mock).mockImplementation(mockedShowWhen);
    (MyFlightCardActions as unknown as jest.Mock).mockImplementation(
      MockedMyFlightCardActions
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders successfully', () => {
    // Arange
    // Act
    const { baseElement } = renderWithTheme(
      <FlightCard data={mockedMyFlight} />
    );
    // Assert
    expect(baseElement).toMatchSnapshot();
    expect(MockedMyFlightCardActions).toBeCalledWith(
      { data: mockedMyFlight },
      {}
    );
  });

  describe('when limited data passed', () => {
    test('renders successfully', () => {
      // Arange
      // Act
      const { baseElement } = renderWithTheme(
        <FlightCard data={mockedMyFlight3} />
      );
      // Assert
      expect(baseElement).toMatchSnapshot();
      expect(MockedMyFlightCardActions).toBeCalledWith(
        { data: mockedMyFlight3 },
        {}
      );
    });
  });

  describe('when screen is lower than sm breakpoint', () => {
    test('renders successfully', () => {
      // Arange
      const mockedBreakpoints = {
        down: { sm: true },
      };
      const mockedUseThemeBreakpoints = jest.fn(() => mockedBreakpoints);
      (useThemeBreakpoints as unknown as jest.Mock).mockImplementation(
        mockedUseThemeBreakpoints
      );
      // Act
      const { baseElement } = renderWithTheme(
        <FlightCard data={mockedMyFlight} />
      );
      // Assert
      expect(baseElement).toMatchSnapshot();
      expect(MockedMyFlightCardActions).toBeCalledWith(
        { data: mockedMyFlight },
        {}
      );
    });
  });
});
