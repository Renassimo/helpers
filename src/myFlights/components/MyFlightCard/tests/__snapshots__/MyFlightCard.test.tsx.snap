// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFlightCard renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-urj5pv-MuiGrid-root"
      >
        <img
          src="photoUrl1"
          width="100%"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1hbdv7e-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Origin
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            originName1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Destination
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            destinationName1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Flight number
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            flightNumber1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Alt. flight number
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            altFlightNumber1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Alt. airline
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            altAirline1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Distance, km
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            1111
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Seat number
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            seatNumber1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Airplane name
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            airplaneName1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            First flight
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            firstFlight1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Age at the flight
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            age1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Serial
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            cn1
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-1 css-1dfagy3-MuiGrid-root"
      >
        <div>
          MockedMyFlightCardActions - 
          fl1
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`MyFlightCard when limited data passed renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-urj5pv-MuiGrid-root"
      >
        <img
          src="/images/plane.png"
          width="100%"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1hbdv7e-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Origin
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            originName3
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Destination
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            destinationName3
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Distance, km
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            3333
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-1 css-1dfagy3-MuiGrid-root"
      >
        <div>
          MockedMyFlightCardActions - 
          fl3
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`MyFlightCard when screen is lower than sm breakpoint renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-urj5pv-MuiGrid-root"
      >
        <img
          src="photoUrl1"
          width="100%"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1hbdv7e-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Origin
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            originName1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Destination
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            destinationName1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Flight number
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            flightNumber1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Alt. flight number
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            altFlightNumber1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Alt. airline
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            altAirline1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Distance, km
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            1111
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Seat number
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            seatNumber1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Airplane name
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            airplaneName1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            First flight
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            firstFlight1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Age at the flight
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            age1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Registration
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            registration1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Serial
            : 
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            cn1
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-1 css-1dfagy3-MuiGrid-root"
      >
        <div>
          MockedMyFlightCardActions - 
          fl1
        </div>
      </div>
    </div>
  </div>
</body>
`;
